services/vision/mlvision/detector.go

```diff
@@ -107,6 +102,134 @@ func attemptToBuildDetector(mlm mlmodel.Service) (objectdetection.Detector, erro
 	}, nil
 }
 
+// findDetectionTensors finds the tensors that are necessary for object detection
+// the returned tensor order is location, category, score.
+func findDetectionTensors(outMap ml.Tensors) (*tensor.Dense, *tensor.Dense, *tensor.Dense, error) {
+	locations, okLoc := outMap["location"]
+	categories, okCat := outMap["category"]
+	scores, okScores := outMap["score"]
+	if okLoc && okCat && okScores { // names are as expected
+		return locations, categories, scores, nil
+	}
+	return guessDetectionTensors(outMap)
+}
+
+// guessDetectionTensors is a hack-y function meant to find the correct detection tensors if the tensors
+// were not given the expected names, or have no metadata. This function should succeed
+// for models built with the viam platform.
+func guessDetectionTensors(outMap ml.Tensors) (*tensor.Dense, *tensor.Dense, *tensor.Dense, error) {
```

Comment:
This private function is where I try to stuff all of our guesses for how tensors are shaped for detections. It's flaky and won't always work. We probably need the re-map tensor name attributes in the tflite_cpu config to really get rid of this. In the meantime, this should take into account any model that we produce with our training tool even if it doesn't include metadata. FYI @tahiyasalam